CONTRIBUTING.md

```diff
@@ -2,7 +2,52 @@
 
 This document provides a collection of rules and guidelines that need to be adhered to when working with or contributing to FluidMS.
 
-## Commit guidelines
+## Git workflow
+
+Before you submit a pull request, please create an issue first so we can discuss the need for this specific feature and clarify the details of the implementation before you invest your time in any work that is potentially wasted.
+
+### Creating a GitHub issue
+
+For issues, please adhere to the following formatting of the issue title:
+
+```
+[Type] <description of the issue>
```

Comment:
This does not match the guidelines for the commit titles. Is this intended?

Reply:
No. You’re right, we should align this to the commit guidelines.